vuu-ui/packages/vuu-shell/src/density-switch/DensitySwitch.tsx

```diff
@@ -1,28 +1,28 @@
 import { Dropdown, SelectionChangeHandler } from "@heswell/salt-lab";
-import { DEFAULT_DENSITY, Density } from "@salt-ds/core";
-import { HTMLAttributes, useCallback } from "react";
+import { Density } from "@salt-ds/core";
+import { HTMLAttributes, useCallback, useContext } from "react";
 import cx from "classnames";
+import { ThemeContext } from "../theme-provider";
 
 const classBase = "vuuDensitySwitch";
 
 const densities:Density[] = ["high", "medium", "low", "touch"];
+const DEFAULT_DENSITY = "high";
 
 export interface DensitySwitchProps
   extends Omit<HTMLAttributes<HTMLDivElement>, "onChange"> {
-    defaultDensity?: Density;
-    density?: Density;
-    onDensityChange: (density: Density) => void;
+  defaultDensity?: Density;
+  density?: Density;
 }
 
 export const DensitySwitch = ({
   className: classNameProp,
   defaultDensity=DEFAULT_DENSITY,
-  onDensityChange
 }:DensitySwitchProps) => {
-
+  const { setDensity } = useContext(ThemeContext)
```

Comment:
while this is fine, I think it makes for a nicer API to export a specialised hook (in the ThemeProvider) for consumers of density
```
const {setDensity} = useDensity()
```

So client do not need to reference the ThemeContext object at all.